src/utils.rs

```diff
@@ -58,6 +64,59 @@ pub fn find_content_components<P: AsRef<Path>>(path: P) -> Vec<String> {
     components
 }
 
+fn utime<P: AsRef<Path>>(path: P, accessed: &SystemTime, modified: &SystemTime) -> Result<()> {
+    extern {
+        fn futimens(fd: c_int, times: *const timespec) -> c_int;
+    }
+
+    let path_str = try!(CString::new(path.as_ref().as_os_str().as_bytes()));
+
+    let accessed_since_epoch = accessed.duration_since(UNIX_EPOCH)?;
+    let modified_since_epoch = modified.duration_since(UNIX_EPOCH)?;
+
+    let atime = timespec {
+        tv_sec: accessed_since_epoch.as_secs() as time_t,
+        tv_nsec: accessed_since_epoch.subsec_nanos() as c_long
+    };
+    let mtime = timespec {
+        tv_sec: modified_since_epoch.as_secs() as time_t,
+        tv_nsec: modified_since_epoch.subsec_nanos() as c_long
+    };
+    let times = [atime, mtime];
+
+    let file = File::open(path)?;
+
+    let ret = unsafe { futimens(file.as_raw_fd(), times.as_ptr()) };
+
+    if ret == 0 {
+        Ok(())
+    } else {
+        bail!(IOError::last_os_error())
+    }
+}
+
+
+/// Copy file if size or modification time is different
+pub fn copy_file_if_modified<P: AsRef<Path>>(source: P, target: P) -> Result<()> {
+    let target = target.as_ref();
+    let source = source.as_ref();
+    let source_metadata = metadata(source)?;
+
+    if target.exists() {
+        let target_metadata = metadata(target)?;
+
+        if target_metadata.len() == source_metadata.len() &&
+            target_metadata.modified()? == source_metadata.modified()? {
+            return Ok(())
+        }
+    }
+
+    copy(&source, &target)?;
+    let accessed_time = source_metadata.accessed()?;
+    let modified_time = source_metadata.modified()?;
+    utime(target, &accessed_time, &modified_time)
```

Comment:
I see, that might not be the best way to do it then if it requires libc and unsafe code.

We get the file path that triggered the notify in the `serve` command so I think a simple method `Site::copy_static_file<P: AsRef<Path>>(path: P) -> Result<()>` would be the easiest. Have to see how notify would handle things like copying several files at once but I believe it would work and be more efficient than going through each file